### PR TITLE
Parse patch

### LINKED ISSUE #20: Missing index.st (search plugin breaks parsing titles with `"` or `\`)
- [x] I have searched the [issues](https://github.com/pelican-plugins/search/issues?q=is%3Aissue) (including closed ones) and believe that this is not a duplicate.

## Issue
When and how does the index get built? On two systems, Windows and Ubuntu, I've followed the instructions and verfied that stork is installed and in PATH.

```
$ stork --version
Stork 1.6.0
```

...then launched pelican with `pelican --autoreload --listen`, and in the resultant preview browser window typed text in the search box. 

The web page says _"Error! Check the browser console."_. 
Browser console says _"Uncaught (in promise) undefined "_.
Shell console says:
```
Done: Processed 69 articles, 0 drafts, 0 hidden articles, 1 page, 0 hidden pages and 0 draft pages in 8.83 seconds.
Unable to find `/search-index.st` or variations:
/search-index.st.html
/search-index.st/index.html
/search-index.st 
```
I've searched the file system for those files and they don't exist, so it looks like the index is not being built. How do I test and/or ensure the index is being built?

### PR DESCRIPTION
**Pull request by maphew**
# Pull Request Checklist

Resolves: #20 

- [x] Conformed to **code style guidelines** by running appropriate linting tools
- [ ] Updated **documentation** for changed code

<s>Note: contributing.md doesn't say anything about coding style or lint tools. I used VS-Code python default.</s> I was looking at wrong browser tab.

No doc change needed.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1 +1,3 @@
 poetry.lock
+
+__pycache__/
```
Comment (justinmayer) [inline .gitignore]: Changes unrelated to the issue at hand are best submitted in separate PRs. In this case, however, I'd refrain because these kinds of ignore directives are better placed in a `~/.gitignore` file on your own workstation, so that they will apply to **all** projects that you interact with. In summary, please remove this change from this PR. 😊

File: pelican/plugins/search/search.py [hunk 2]
```diff
@@ -15,6 +15,7 @@
 import subprocess
 
 from jinja2.filters import do_striptags as striptags
+from json import dumps
```
Comment (justinmayer) [inline pelican/plugins/search/search.py]: Did you [set up your plugin dev environment](https://docs.getpelican.com/en/latest/contribute.html#plugin-development) and then run the linters locally before committing and pushing? I only ask because CI is showing a linter failure due to disordered import statements.

**Review by justinmayer — CHANGES_REQUESTED** [review]
Hey Matt. Many thanks for the pull request. I was about to merge #15 and then this PR when I realized that they both modify how the `title` is handled. I think it's best if I merge #15 first, and then perhaps you can modify this PR appropriately. After making your follow-up changes to resolve the conflict, if you would be so kind as to ensure that titles that contain double-quotation-marks are still handled properly (i.e., the issue that #15 was submitted to fix), that would be much appreciated!

File: pelican/plugins/search/search.py (hunk 2 above)
Comment (maphew) [reply pelican/plugins/search/search.py]: I have now (and thanks for the coaching!) but something in that recipe is seriously at odds with my system. Now every commit tries to reinstall poetry, invoke, etc., then encounters a compatbility error and aborts. I'll follow up with a seperate issue after I've had a chance to try and understand what's going on. For now I'll just edit the file manually via Github.

File: .gitignore (hunk 1 above)
Comment (maphew) [reply .gitignore]: cool! I didn't know a person could do that.

**Comment by maphew** [comment]
Ok, I don't know what happened. I didn't intentionally close this PR. Maybe Github did it automatically when I synced the fork with upstream to include #15. However maybe that's for the best since this was going sideways anyway.
